### PR TITLE
Report local versions for compatible packages

### PR DESCRIPTION
**Pull request by DCKcode**
This PR adds some functionality to make sure locally installed versions are available in the report as well. This information should be included for all packages that use Gomod.

Some other minor changes:
* Ability to handle upstream sources that are not internal GitLab, GitHub, and Google's Gerrit
* Changing the case of how "license" is spelled in the resulting report to be compatible with other internal tooling we use

### CODE REVIEW

File: report/report.go [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
 
 const (
-	GITHUB = "github"
-	GITLAB = "gitlab"
-	GERRIT = "gerrit"
+	GITHUB  = "github"
+	GITLAB  = "gitlab"
+	GERRIT  = "gerrit"
+	UNKNOWN = "unknown/other"
 )
```
Comment (complexspaces) [inline report/report.go:19]: Should we add "GoModuleCache" to the list while we are here, or is it possible we aren't using the Google-hosted package caches for these?

File: report/report.go [hunk 2]
```diff
@@ -83,19 +83,17 @@ func (g Generator) reportObjFromDependency(dep models.Dependency) (*models.Repor
...
-		}
 	case GERRIT:
 		reportObject, err = versioncontrol.ReportObjFromGerrit(dep, g.request)
 		if err != nil {
 			return nil, errors.Wrapf(err, "unable to generate reportObject from dependency %s", dep.Name)
 		}
 	default:
-		log.Println("Unable to determine repo source for ", dep.Name)
-		return nil, fmt.Errorf("unable to determine repo source for %s", dep.Name)
+		reportObject, err = versioncontrol.ReportObjGeneric(dep)
```
Comment (complexspaces) [inline report/report.go:94]: Nit: It might be good to leave a comment at the top of the `switch` to document when you should do something special-cased vs using the generic handler.

**Review by complexspaces — COMMENTED** [review]
This looks good to me for the most part, I just had a few small comments.

File: report/report.go (hunk 1 above)
Comment (DCKcode) [reply report/report.go:19]: What would "GoModuleCache" mean here? I may not understand what you're going for 🙂 

This enum reflects where the package _source_ is hosted (GitHub, and Google's Gerrit for the vast majority, internal GitLab for some, but exceptions do exist in codebases). So it's not about where it fetches the actual module from, but what the original Git source is.

File: report/report.go (hunk 2 above)
Comment (DCKcode) [reply report/report.go:94]: Added in https://github.com/1Password/dep-report/pull/40/commits/5dfb7c07167ed7f23fdbef12e78ccdd941decf94 ✅

**Comment by dannywieser-1p** [comment]
@DCKcode will this have any impact on the [common failures we get with a missing mapping for a new dependency?](https://github.com/1Password/dep-report/pull/39). Would those fall into the unknown bucket?

**Review by dannywieser-1p — APPROVED** [review]
These changes look good to me - I had one question and I'm also curious if you've done a test run of this against the current b5 codebase - the release pipeline uses this tool daily.

**Comment by DCKcode** [comment]
> @DCKcode will this have any impact on the [common failures we get with a missing mapping for a new dependency?](https://github.com/1Password/dep-report/pull/39). Would those fall into the unknown bucket?

I believe so. While running against our internal `op` repository I encountered another edge case. This is my way of resolving the issue so the tool doesn't fail prematurely in the future.

> These changes look good to me - I had one question and I'm also curious if you've done a test run of this against the current b5 codebase - the release pipeline uses this tool daily.

Yes - it creates updated reports for B5 as well, all of its go.mod files.

File: report/report.go (hunk 1 above)
Comment (complexspaces) [reply report/report.go:19]: Ah alright, that makes sense then.

**Review by complexspaces — APPROVED** [review]

**Comment by dannywieser-1p** [comment]
> > @DCKcode will this have any impact on the [common failures we get with a missing mapping for a new dependency?](https://github.com/1Password/dep-report/pull/39). Would those fall into the unknown bucket?
> 
> I believe so. While running against our internal `op` repository I encountered another edge case. This is my way of resolving the issue so the tool doesn't fail prematurely in the future.
> 
> > These changes look good to me - I had one question and I'm also curious if you've done a test run of this against the current b5 codebase - the release pipeline uses this tool daily.
> 
> Yes - it creates updated reports for B5 as well, all of its go.mod files.

I like that it won't fail - but I guess we'll be missing notifications about a missing mapping. Is the information reported in these cases incomplete?

**Comment by DCKcode** [comment]
In those cases it will only contain information about the local version. It won't be able to look up the latest _available_ version remotely.

This is actually already the case for the GitLab provider we have currently 🙂
